interpreter/cling/lib/Interpreter/DynamicLibraryManager.cpp

```diff
@@ -65,13 +65,28 @@ namespace cling {
     m_SearchPaths.push_back({".", /*IsUser*/true});
   }
 
-  std::string
-  DynamicLibraryManager::lookupLibInPaths(llvm::StringRef libStem) const {
+  std::string DynamicLibraryManager::lookupLibInPaths(
+      llvm::StringRef libStem, const std::string& RPATH /*=""*/,
+      const std::string& RUNPATH /*=""*/) const {
+
     llvm::SmallVector<SearchPathInfo, 128> Paths;
+
+    // RPATH, LibSearchPath, RUNPATH
+    llvm::SmallVector<llvm::StringRef, 2> RPaths;
+    SplitPaths(RPATH, RPaths, utils::kPruneNonExistant, platform::kEnvDelim,
+               false);
+    for (auto& P : RPaths)
+      Paths.push_back({P.str(), /*IsUser*/ true});
+    // LibSearchPath, m_SearchPaths
     for (const std::string &P : m_Opts.LibSearchPath)
       Paths.push_back({P, /*IsUser*/true});
-
     Paths.append(m_SearchPaths.begin(), m_SearchPaths.end());
+    // RUNPATH
+    llvm::SmallVector<llvm::StringRef, 2> RunPaths;
+    SplitPaths(RUNPATH, RunPaths, utils::kPruneNonExistant, platform::kEnvDelim,
+               false);
+    for (auto& P : RunPaths)
+      Paths.push_back({P.str(), /*IsUser*/ true});
```

Comment:
Why is the order of priority

1. RPATH
2. user-provided paths and LD_LIBRARY_PATH
3. RUNPATH?

Maybe worth a comment?

Reply:
We have now a comment with a link to the description of the priorities for the static linker.